octopoes/tools/rename-origin-method.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import json
+import logging
+from typing import Any
+
+import click
+from xtdb_client import XTDBClient
+
+logger = logging.getLogger(__name__)
+
+
+@click.group(
+    context_settings={
+        "help_option_names": ["-h", "--help"],
+        "max_content_width": 120,
+        "show_default": True,
+    }
+)
+@click.option("-n", "--node", default="0", help="XTDB node")
+@click.option(
+    "-u",
+    "--url",
+    default="http://localhost:3000",
+    help="XTDB server base url",
+)
+@click.option(
+    "-t",
+    "--timeout",
+    type=int,
+    default=5000,
+    help="XTDB request timeout (in ms)",
+)
+@click.option("-v", "--verbosity", count=True, help="Increase the verbosity level")
+@click.pass_context
+def cli(ctx: click.Context, url: str, node: str, timeout: int, verbosity: int):
+    verbosities = [logging.WARN, logging.INFO, logging.DEBUG]
+    try:
+        if verbosity:
+            logging.basicConfig(level=verbosities[verbosity - 1])
+    except IndexError:
+        raise click.UsageError("Invalid verbosity level (use -v, -vv, or -vvv)")
+
+    client = XTDBClient(url, node, timeout)
+    logger.info("Instantiated XTDB client with endpoint %s for node %s", url, node)
+
+    ctx.ensure_object(dict)
+    ctx.obj["client"] = client
+
+
+QY = '{:query {:find [(pull ?var [*])] :where [[?var :type "Origin"][?var :origin_type "observation"][?var :method]]}}'
+
+
+def method_query(method: str):
+    if method:
+        return f"""{{
+:query {{
+    :find [(pull ?var [*])] :where [
+            [?var :type "Origin"]
+            [?var :origin_type "observation"]
+            [?var :method "{method}"]
+        ]
+    }}
+}}
+"""
+    else:
+        return QY
+
+
+@cli.command(help="List observation origins based on method")
+@click.argument("method", required=False)
+@click.pass_context
+def LIST(ctx: click.Context, method: str):
+    origins = ctx.obj["client"].query(method_query(method))
+    if not origins:
+        raise click.UsageError("No targets found")
```

Comment:
You might want to consider warning the user using `click.echo` with the `err=True` parameter.

A `UsageError` would imply that the user misused the program, but in my opinion, this seems more like a reasonable runtime error. If it indeed fits the context of user misuse, then the `UsageError` is fine

Reply:
Debatable, I am fine either way. The intention was to throw an error because no targets are found, perhaps because the user typed-in an "invalid" method...

Reply:
Clear. Both is fine with me, I'll leave it up to you!